### PR TITLE
Add semver module

### PR DESCRIPTION
**Pull request by DoronCohen**
We need this module available in other projects. I lost track to where should we push this. `master` / one of the `wekapp` branches. I will also need this in a released package and I saw some effort to release a `0.4.0` what is the status of that?

### CODE REVIEW

File: easypy/semver.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+            major=int(major),
+            minor=int(minor),
+            patch=int(patch) if patch is not None else None,
+            build=int(build) if build is not None else None,
+            tag="" if tag is None else str(tag),
+        )
+
+    @property
+    def __dict__(self):
+        return super().__dict__
```
Comment (idanarye) [inline easypy/semver.py]: Why is this necessary?

File: easypy/semver.py [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+    def _to_tuple(self):
+        return (self.major, self.minor, self.patch or 0, self.build or 0)
+
+    def __eq__(self, other):
+        assert isinstance(other, self.__class__)
+        return self._to_tuple() == other._to_tuple() and self.tag == other.tag
+
+    def __lt__(self, other):
+        assert isinstance(other, self.__class__)
+        return self._to_tuple() < other._to_tuple() or \
```
Comment (YuvalEvron) [inline easypy/semver.py]: You should add parenthesis to make this statement clear.

File: easypy/semver.py [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+            self._to_tuple() == other._to_tuple() and self.tag < other.tag
+
+    def __gt__(self, other):
+        assert isinstance(other, self.__class__)
+        return self._to_tuple() > other._to_tuple() or \
```
Comment (YuvalEvron) [inline easypy/semver.py]: Same

File: easypy/semver.py (hunk 1 above)
Comment (koreno) [reply easypy/semver.py]: seems like it doesn't work anyway. I'll remove this
